src/bindata.cr

```diff
@@ -163,6 +164,24 @@ abstract class BinData
         {% end %}
       {% end %}
 
+      {% if REMAINING.size > 0 %}
+        {% if REMAINING[0][2] %}
+          %onlyif = ({{REMAINING[0][2]}}).call
+          if %onlyif
+        {% end %}
+        %buf = Bytes.new io.size - io.pos
+        io.read_fully %buf
+        @{{REMAINING[0][1]}} = %buf
+        {% if REMAINING[0][2] %}
+          end
+        {% end %}
+        {% if REMAINING[0][3] %}
+          if !({{REMAINING[0][3]}}).call
+            raise VerificationException.new "Failed to verify reading #{{{REMAINING[0][0]}}} at {{@type}}.{{REMAINING[0][1]}}"
+          end
+        {% end %}
+      {% end %}
```

Comment:
I don't like indexing into REMAINING with [0][idx], but I don't know how to do variable assignment with macros... Is there another way to do this? There's no reason that this should be rested..

Reply:
Yeah it's all a bit of a compiler hack to get maximum parsing speed as everything is unwrapped at compile time.
There is no way to have global macro variable assignment except you can update an array.

In terms of `[0][idx]` you can use named tuples in macros, which is something I would change if refactoring the library

```
{% REMAINING << {type: "bytes", name: name.id, onlyif: onlyif, verify: verify} %}
```

Then in the parsing code everything would be a little clearer

Reply:
Feel free to see if the named tuple version works. Might as well start using that and I can update the rest of the lib at some point in the future